### PR TITLE
eos-download-image: handle missing manifest gracefully

### PR DESCRIPTION
**Pull request by wjt**
Without this change, running `eos-download-image --product esodvd` gives
you a long traceback followed by:

    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

With this change, you get a more human-readable error describing the
actual cause:

    Couldn't fetch manifest for product esodvd

    403 Client Error: Forbidden for url: https://d1anzknqnc1kmb.cloudfront.net/releases-esodvd-3.json

Found while QAing https://phabricator.endlessm.com/T18653

### CODE REVIEW

File: eos-tech-support/eos-download-image [hunk 1]
```diff
@@ -298,7 +298,18 @@ class EosDownloadImage(object):
...
+        response = self.session.get(manifest_url)
+
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            log("Couldn't fetch manifest for product", self.args.product)
+            log(e)
+            sys.exit(1)
+
+        print(response.headers)
```
Comment (ramcq) [inline eos-tech-support/eos-download-image]: Is this desirable in normal operation?

File: eos-tech-support/eos-download-image [hunk 2]
```diff
@@ -298,7 +298,18 @@ class EosDownloadImage(object):
...
+        response.raise_for_status()
```
Comment (ramcq) [inline eos-tech-support/eos-download-image]: Seems redundant with the above call in the try: block?

File: eos-tech-support/eos-download-image (hunk 1 above)
Comment (wjt) [inline eos-tech-support/eos-download-image]: Ah! You spotted my, uh, check that the reviewers are reviewing. Definitely not testing code left in by accident.

File: eos-tech-support/eos-download-image [hunk 3]
```diff
@@ -298,8 +298,16 @@ class EosDownloadImage(object):
...
+        response = self.session.get(manifest_url)
 
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            log("Couldn't fetch manifest for product", self.args.product)
+            log(e)
+            sys.exit(1)
+
+        manifest = response.json()
```
Comment (wjt) [inline eos-tech-support/eos-download-image]: Updated to add this line, without which, of course, nothing works. Apparently I tested the error path, but not the happy path.

**Comment by wjt** [comment]
Added a patch on here to refresh the keyring.

**Pull request merged**